src/useEffect-example/index.js

```diff
@@ -4,14 +4,21 @@ function App() {
   const [isOn, setIsOn] = useState(false);
   const [timer, setTimer] = useState(0);
 
-  useEffect(() => {
-    const interval = setInterval(
-      () => isOn && setTimer(timer + 1),
-      1000,
-    );
-
-    return () => clearInterval(interval);
-  }, []);
+  useEffect(
+    () => {
+      let interval;
+
+      if (isOn) {
+        interval = setInterval(
+          () => setTimer(timer => timer + 1),
+          1000,
+        );
+      }
+
+      return () => clearInterval(interval);
```

Comment:
Hi @rwieruch , in case that `isOn === false`, this will do `clearInterval(undefined)`. It doesn't break anything, but it would be IMHO nicer to only run `clearInterval` for `isOn === true`.